File: src/lib/tauri-api/repository-relationships.ts
import { invokeCommand } from './client';
import type {
  CreateRepositoryRelationshipInput,
  RelevantRepositoriesSuggestionResult,
  RepositoryRelationshipsResult,
  SuggestRelevantRepositoriesInput,
  UpdateRepositoryRelationshipInput,
} from '../../types/repository-relationship';

export function listRepositoryRelationships(): Promise<RepositoryRelationshipsResult> {
  return invokeCommand<RepositoryRelationshipsResult>('list_repository_relationships');
}

export function createAppRepositoryRelationship(
  input: CreateRepositoryRelationshipInput,
): Promise<RepositoryRelationshipsResult> {
  return invokeCommand<RepositoryRelationshipsResult>('create_app_repository_relationship', { input });
}

export function updateAppRepositoryRelationship(
  input: UpdateRepositoryRelationshipInput,
): Promise<RepositoryRelationshipsResult> {
  return invokeCommand<RepositoryRelationshipsResult>('update_app_repository_relationship', { input });
}

export function deleteAppRepositoryRelationship(
  relationshipId: string,
): Promise<RepositoryRelationshipsResult> {
  return invokeCommand<RepositoryRelationshipsResult>('delete_app_repository_relationship', { relationshipId });
}

export function suggestRelevantRepositoriesForTask(
  input: SuggestRelevantRepositoriesInput,
): Promise<RelevantRepositoriesSuggestionResult> {
  return invokeCommand<RelevantRepositoriesSuggestionResult>('suggest_relevant_repositories_for_task', { input });
}

File: src-tauri/src/commands/repository_relationships.rs
use tauri::State;

use crate::models::{
    CreateRepositoryRelationshipInput, RelevantRepositoriesSuggestionResult,
    RepositoryRelationshipsResult, SuggestRelevantRepositoriesInput,
    UpdateRepositoryRelationshipInput,
};
use crate::services::repository_relationship_service;
use crate::state::AppState;

#[tauri::command]
pub fn list_repository_relationships(
    state: State<'_, AppState>,
) -> Result<RepositoryRelationshipsResult, String> {
    repository_relationship_service::list_repository_relationships(&state)
}

#[tauri::command]
pub fn create_app_repository_relationship(
    state: State<'_, AppState>,
    input: CreateRepositoryRelationshipInput,
) -> Result<RepositoryRelationshipsResult, String> {
    repository_relationship_service::create_app_repository_relationship(&state, input)
}

#[tauri::command]
pub fn update_app_repository_relationship(
    state: State<'_, AppState>,
    input: UpdateRepositoryRelationshipInput,
) -> Result<RepositoryRelationshipsResult, String> {
    repository_relationship_service::update_app_repository_relationship(&state, input)
}

#[tauri::command]
pub fn delete_app_repository_relationship(
    state: State<'_, AppState>,
    relationship_id: String,
) -> Result<RepositoryRelationshipsResult, String> {
    repository_relationship_service::delete_app_repository_relationship(&state, &relationship_id)
}

#[tauri::command]
pub fn suggest_relevant_repositories_for_task(
    state: State<'_, AppState>,
    input: SuggestRelevantRepositoriesInput,
) -> Result<RelevantRepositoriesSuggestionResult, String> {
    repository_relationship_service::suggest_relevant_repositories_for_task(&state, input)
}
